### PR TITLE
Prepare for Renovate with reusable workflows

### PR DESCRIPTION
**Pull request by ghostwriter**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Adding Renovate to Laminas, Mezzio repositories. https://gist.github.com/weierophinney/49ca8ae2b916d898dd3d4162b26ff41b

- Configure minimum supported PHP version
- Update composer dependencies
- Update to reusable `Continuous Integration` workflow
- Update to reusable `Automatic Releases` workflow

### CODE REVIEW

**Review by Ocramius — APPROVED** [review]

**Review by Ocramius — CHANGES_REQUESTED** [review]
Redis is needed here: some rollback necessary 🤔

**Comment by Ocramius** [comment]
Failures unrelated to patch - I think the setup of CI on this repo is simply borked

**Comment by boesing** [comment]
@Ocramius its not borked. This PR removes the docker service of `redis` and thus tests are failing.
If you remove tires from a car, it won't drive anymore and its not because its borked ;-)

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -4,42 +4,8 @@ on:
...
-      - name: Gather CI configuration
-        id: matrix
-        uses: laminas/laminas-ci-matrix-action@v1
-
-  qa:
-    name: QA Checks
-    needs: [matrix]
-    runs-on: ${{ matrix.operatingSystem }}
-    services:
-      redis:
```
Comment (boesing) [inline .github/workflows/continuous-integration.yml]: redis service is required to make tests pass

**Review by boesing — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
Sorry, SNAFU on my end - indeed, I forgot the scope of the problems here, which I already discussed with @ghostwriter elsewhere. In practice, reusable workflows don't support services :|

**Review by Ocramius — APPROVED** [review]
Thanks @ghostwriter!

**Pull request merged**